client/dashboard/me/profile-personal-details/email-section.tsx

```diff
@@ -0,0 +1,159 @@
+import { userSettingsQuery, cancelPendingEmailChangeMutation } from '@automattic/api-queries';
+import { useMutation, useSuspenseQuery, useQueryClient } from '@tanstack/react-query';
+import { __experimentalInputControl as InputControl, Button } from '@wordpress/components';
+import { createInterpolateElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Icon, info, check } from '@wordpress/icons';
+import emailValidator from 'email-validator';
+import { useState, useEffect, useCallback } from 'react';
+import './style.scss';
+
+interface EmailSectionProps {
+	value: string;
+	onChange: ( value: string ) => void;
+	disabled?: boolean;
+}
+
+export default function EmailSection( { value, onChange, disabled = false }: EmailSectionProps ) {
+	const { data: userData } = useSuspenseQuery( userSettingsQuery() );
+	const queryClient = useQueryClient();
+
+	const { mutate: cancelPendingEmail, isPending: isCancelPending } = useMutation( {
+		...cancelPendingEmailChangeMutation(),
+		onSuccess: () => {
+			onChange( userData.user_email || '' );
+			// Invalidate user settings to refetch updated data on cancel
+			queryClient.invalidateQueries( userSettingsQuery() );
```

Comment:
Please move invalidation to the query layer, in the `cancelPendingEmailChangeMutation()`'s onSuccess itself. The idea is that components shouldn't care about invalidations; it should be handled by the queries.